opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/websocket/TracingChannelInterceptor.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.cloud.websocket;
+
+import org.springframework.messaging.simp.SimpMessageType;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.support.ChannelInterceptorAdapter;
+import org.springframework.messaging.support.ExecutorChannelInterceptor;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.web.socket.messaging.SubProtocolWebSocketHandler;
+import org.springframework.web.socket.messaging.WebSocketAnnotationMethodMessageHandler;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.Tracer.SpanBuilder;
+import io.opentracing.propagation.Format;
+import io.opentracing.tag.Tags;
+
+/**
+ * This class implements a {@link ExecutorChannelInterceptor} to instrument the websocket
+ * communications using an OpenTracing Tracer.
+ *
+ */
+public class TracingChannelInterceptor extends ChannelInterceptorAdapter implements ExecutorChannelInterceptor {
+
+    /**
+     * The STOMP simple destination.
+     */
+    protected static final String SIMP_DESTINATION = "simpDestination";
+
+    /**
+     * The STOMP simple message type, values defined in enum {@link SimpMessageType}.
+     */
+    protected static final String SIMP_MESSAGE_TYPE = "simpMessageType";
+
+    /**
+     * Indicates that the destination is unknown.
+     */
+    private static final String UNKNOWN_DESTINATION = "Unknown";
+
+    /**
+     * Header name used to carry the current {@link Span} from the initial preSend phase to
+     * the beforeHandle phase.
+     */
+    protected static final String OPENTRACING_SPAN = "opentracing.span";
+
+    private Tracer tracer;
+    private String spanKind;
+
+    public TracingChannelInterceptor(Tracer tracer, String spanKind) {
+        this.tracer = tracer;
+        this.spanKind = spanKind;
+    }
+
+    @Override
+    public Message<?> preSend(Message<?> message, MessageChannel channel) {
+        if (SimpMessageType.MESSAGE.equals(message.getHeaders().get(SIMP_MESSAGE_TYPE))) {
+            SpanContext context = null;
```

Comment:
This can be moved into `if (spanKind == Tags.SPAN_KIND_SERVER) {` if you move it after span creation.

Reply:
It would be better to split this method for server and client.